backend/service/topology/cache.go

```diff
@@ -27,22 +36,27 @@ func (c *client) acquireTopologyCacheLock(ctx context.Context) {
 	}
 	defer conn.Close()
 
-	advisoryLockId := convertLockIdToAdvisoryLockId(topologyCacheLockId)
+	// catch signals, shutdown caching by canceling the context, and release the advisory lock.
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	go func() {
+		<-sigc
+		c.log.Info("Caught shutdown signal, shutting down topology caching and releasing advisory lock")
+		ticker.Stop()
+		ctxCancelFunc()
+		c.unlockAdvisoryLock(context.TODO(), conn, advisoryLockId)
+	}()
```

Comment:
i would push the signal catching up another level, and use the incoming context for cancellation.

eventually the gateway itself will catch the signal and propagate the event, see https://github.com/lyft/clutch/issues/12 graceful shutdown

Reply:
Doesn't that mean i need to pass the cancel func to `acquireTopologyCacheLock` in addition to the context? or is their an easier way of doing this?